src/sentry/lang/javascript/processor.py

```diff
@@ -970,21 +1163,39 @@ def process_frame(self, processable_frame, processing_task):
         ):
             return
 
+        # We get the debug_id for file in this frame, if any.
+        debug_id = self.abs_path_debug_id.get(frame["abs_path"])
+
         # The 'sourceview' is used for pre/post and 'context_line' frame expansion.
         # Here it's pointing to the minified source, however the variable can be shadowed with the original sourceview
-        # (or 'None' if the token doesnt provide us with the 'context_line') down the road.
-        # TODO: add support for debug ids.
-        sourceview = self.get_or_fetch_sourceview(url=frame["abs_path"])
-        sourcemap_cache = self.get_or_fetch_sourcemap_cache(url=frame["abs_path"])
+        # (or 'None' if the token doesn't provide us with the 'context_line') down the road.
+        sourceview, resolved_sv_with_debug_id = self.get_or_fetch_sourceview(
+            url=frame["abs_path"],
+            debug_id=debug_id,
+            source_file_type=SourceFileType.MINIFIED_SOURCE,
+        )
+        # Only if we resolved the minified source with a debug_id we want to try and resolve the sourcemap with the
+        # debug_id.
+        sourcemap_cache, resolved_smc_with_debug_id = self.get_or_fetch_sourcemap_cache(
+            url=frame["abs_path"], debug_id=debug_id if resolved_sv_with_debug_id else None
+        )
 
         # We check for errors once both the minified file and the corresponding sourcemaps are loaded. In case errors
         # happen in one of the two we will detect them and add them to 'all_errors'.
-        errors = self.fetch_by_url_errors.get(frame["abs_path"])
-        if errors is not None:
-            all_errors.extend(errors)
+        url_errors = self.fetch_by_url_errors.get(frame["abs_path"])
+        if url_errors is not None:
+            all_errors.extend(url_errors)
+        debug_id_errors = self.fetch_by_debug_id_errors.get(debug_id)
+        if debug_id_errors is not None:
+            all_errors.extend(debug_id_errors)
 
         # We get the url that we discovered for the sourcemap which we use down in the processing pipeline.
-        sourcemap_url = self.minified_source_url_to_sourcemap_url.get(frame["abs_path"])
+        # In case we resolved the sourcemap with the debug_id, the url will be the debug_id.
```

Comment:
btw, this is a good idea. is this exposed somewhere (except in tests?)

Reply:
We put the resulting `sourcemap_label` (derived from `sourcemap_url`) within the `new_frame`: https://github.com/getsentry/sentry/blob/7ca6d240c1f4e823c895b8fc3cfe43f06482989c/src/sentry/lang/javascript/processor.py#L1247-L1247.